workflow-runtime/src/commonMain/kotlin/com/squareup/workflow1/internal/WorkflowNode.kt

```diff
@@ -96,7 +96,12 @@ internal class WorkflowNode<PropsT, StateT, OutputT, RenderingT>(
   private val eventActionsChannel =
     Channel<WorkflowAction<PropsT, StateT, OutputT>>(capacity = UNLIMITED)
   private var state: StateT
-  private var subtreeStateDidChange: Boolean = true
+
+  // Our state or that of one of our descendants changed.
+  private var subtreeStateDirty: Boolean = true
+
+  // Our state changed.
+  private var selfStateDirty: Boolean = true
```

Comment:
Maybe i missed it in the code, but it looks like we never set this on its own (without also setting `subtreeStateDirty`)? Do we really need it?

Reply:
Yes, lines 349-350 we set them using different conditions when the action is being applied.

```
// Our state changed.
    selfStateDirty = actionApplied.stateChanged
    // Our state changed or one of our children's state changed.
    subtreeStateDirty = aggregateActionApplied.stateChanged
```

Reply:
that's key for drain exclusive because `subtreeStateDirty` is true always on a path all the way to the root no matter what changed in the tree. We use `selfStateDirty` to determine whether the node is exclusive to the previous action applied because otherwise there would never be any other exclusive nodes that we can process actions on.